### ISSUE TITLE
Refactor: Standardize Epsilon Handling for Numerical Stability

### ISSUE THREAD

**Issue by dcruzf** [dcruzf/highFIS#55]
Currently, the highFIS library suffers from inconsistent numerical stability constants (eps) across its core modules:
 * memberships.py: Hardcoded eps=1e-6 as the default in MembershipFunction and _inv_softplus.
 * defuzzifiers.py: Hardcoded eps=1e-8 as the default for log and sum operations.
 * t_norms.py: Dynamically infers epsilon using torch.finfo(terms.dtype).eps.
This inconsistency creates mismatched clipping thresholds during the forward pass. More importantly, hardcoded floats like 1e-8 will immediately cause underflow errors if the user attempts to train the model using Mixed Precision (e.g., torch.float16 or torch.bfloat16), where the minimum representable positive value is much larger (around 5.96e-8 for FP16).
## 🧮 Mathematical Context
In operations like the SoftmaxLogDefuzzifier, we compute \log(\bar{w}_r). If a rule is completely inactive (w_r = 0), the logarithm approaches -\infty, producing NaNs in the computational graph. We mitigate this via:

To guarantee that the gradient flows smoothly from the Defuzzifier down to the Membership layer, \epsilon must be universally agreed upon across the entire neuro-fuzzy pipeline and respect the hardware tensor limits.
## 💡 Proposed Solution
We should standardize epsilon resolution by adopting the dynamic approach (torch.finfo) globally, removing hardcoded defaults.
### Implementation Steps
**1. Update defuzzifiers.py:**
Remove the hardcoded default and extract it dynamically from the input tensor.
```python
class SoftmaxLogDefuzzifier(nn.Module):
    def __init__(self, eps: float | None = None) -> None:
        super().__init__()
        self.eps = eps

    def forward(self, w: Tensor) -> Tensor:
        # Dynamically infer epsilon if not explicitly provided
        eps = self.eps if self.eps is not None else torch.finfo(w.dtype).eps
        log_w = w.clamp(min=eps).log()
        return torch.softmax(log_w, dim=1)

```
**2. Update memberships.py:**
Ensure parameter initialization respects the default PyTorch precision.
```python
class MembershipFunction(nn.Module):
    def __init__(self, eps: float | None = None) -> None:
        super().__init__()
        # Use PyTorch's default dtype to infer epsilon during init
        self.eps = eps if eps is not None else torch.finfo(torch.get_default_dtype()).eps

```
## ✅ Expected Benefits
 * **Mixed Precision Compatibility:** Safely allows highFIS to run in FP16/BF16 without underflow.
 * **Smooth Gradients:** Ensures a uniform clamping floor across the entire network, preventing gradient bottlenecks.
 * **Cleaner API:** Users don't need to guess which epsilon to tweak if numerical instabilities arise; the library intelligently adapts to the tensor type.
